### PR TITLE
Fixed compiling errors and updated keymaps for infinity_chibios

### PR DESCRIPTION
**Pull request by Depariel**
* Fixed `infinity_chibios.h` errors that prevented functions from being used (`KC_` was prefixed to everything)
* Updated keymaps (and added my own) to be compatible with new `infinity_chibios.h`
* Fixed a typo in `config.h` and added `#define PREVENT_STUCK_MODIFIERS`

Fixes some issues from https://github.com/jackhumbert/qmk_firmware/issues/575

* Added matrix support for the latest Infinity 60% board
* No LED support for latest Infinity 60% board yet

Basically added https://github.com/tmk/tmk_keyboard/pull/371 to QMK

### CODE REVIEW

**Comment by xrigau** [comment]
👍 was about to make a similar PR but you saved me the work 😄

**Comment by xrigau** [comment]
just out of curiosity @Depariel how do you flash the firmware? for me when I run `make` it generates a `.hex` file but when I flash that into the keyboard it doesn't really work (back LED doesn't turn off and keystrokes don't do anything)

I was able to convert the `.hex` file to a `.bin` file that I can actually flash into the keyobard (and it works!) using [this hex2bin pyhton script](https://github.com/xrigau/qmk_firmware/blob/c08235f3d477b8654be904e42cd5623073f01b34/keyboards/infinity_chibios/hex2bin.py) (I've [added it to the Makefile](https://github.com/xrigau/qmk_firmware/blob/c08235f3d477b8654be904e42cd5623073f01b34/keyboards/infinity_chibios/Makefile#L73)) and it requires doing the intelhex library `pip install intelhex`

Did you have to do something similar to flash the firmware into the Infinity or am I missing something?
Thanks!

**Comment by F1LT3R** [comment]
I'm also having problems flashing the infinity with hex files (Teensy app). Bin files do work for me as well, using dfu-util.

**Comment by Depariel** [comment]
What I do is run `make dfu-util`, which will give you a `.hex` like you said, but it'll also put a `.bin` in the `.build` hidden directory. I just copy that `.bin` over to my Windows machine and run `kii-dfu.exe` (which you can find on [this page](https://github.com/kiibohd/controller/wiki/Loading-DFU-Firmware)) using that `.bin` as your firmware file.

I have no idea why, but I seem completely unable to flash from the command line in Linux; the above method is the only thing I've found to work for myself so far. I'm not terribly concerned, since I actually do everything from my Windows computer anyway (just remote into my Linux machine to do the compiling and everything), but it's not exactly the most straightforward solution.

**Comment by xrigau** [comment]
Ah that's what I was missing! If I just run `make` it doesn't generate the `.bin` only `.hex`, `.map` and `.elf` but running `make dfu-util` does generate the `.bin` thanks!!

(also seems like the `.build` folder is in the project root instead of in the `infinity_chibios` dir)

**Comment by r2d2rogers** [comment]
I think the issue is permissions, as I am able to flash my ergodox infinity when I use `sudo dfu-util -D ergodox_infinity_r2d2rogers.bin`.  I have not yet found the correct udev rules to make the device accessible without sudo.

**Comment by F1LT3R** [comment]
So I seem to have my left hand infinity stuck in flash mode after trying this.

Steps:

```bash
cd keyboards/infinity_chibios
make dfu-util
cd ../../.build/
dfu-util -D infinity_chibios_default.bin -S mk20...  # device serial
```

**Comment by F1LT3R** [comment]
@xrigau are you building from this PR? I was also finding the `.bin` file in the `../../.build` dir until I checked out this PR.

**Comment by xrigau** [comment]
@F1LT3R I believe the Infinity Ergodox won't work with the firmware in the `infinity_chibios` folder since it's specifically made for the Infinity 60% keyboard - you should probably use the https://github.com/jackhumbert/qmk_firmware/tree/master/keyboards/ergodox/infinity as a starting point. 

And no sorry I was building from the master branch and from my form (but I assumed the results would be the same than doing it from this PR)

**Comment by jackhumbert** [comment]
Cool! Thanks :)
